pkg/instancetype/controller/vm/controller.go

```diff
@@ -138,6 +144,44 @@ func (c *controller) Sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineI
 	return vm, nil
 }
 
+func (c *controller) checkResourcesExist(vm *virtv1.VirtualMachine) error {
+	conditionManager := controllerpkg.NewVirtualMachineConditionManager()
+
+	_, instancetypeErr := c.Find(vm)
+	if instancetypeErr != nil && !conditionManager.HasCondition(vm, virtv1.VirtualMachineInstancetypeNotFound) {
+		conditionManager.UpdateCondition(vm, &virtv1.VirtualMachineCondition{
+			Type:               virtv1.VirtualMachineInstancetypeNotFound,
+			LastTransitionTime: metav1.Now(),
+			Status:             corev1.ConditionTrue,
+			Message:            instancetypeErr.Error(),
+		})
+	}
+
+	if instancetypeErr == nil && conditionManager.HasCondition(vm, virtv1.VirtualMachineInstancetypeNotFound) {
```

Comment:
Is the second conditional required? Or is `RemoveCondition` already a no-op when there is no such condition?

Reply:
Yeah it's not required as RemoveCondition already checks that it's there. I'll remove.